clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -509,7 +509,10 @@ public void addMetric(MetricName metricName, MetricConfig config, MetricValuePro
                                         Objects.requireNonNull(metricValueProvider),
                                         config == null ? this.config : config,
                                         time);
-        registerMetric(m);
+        KafkaMetric existingMetric = registerMetric(m);
```

Comment:
Could we clarify in the javadoc above that if the metric already exists, an exception would be thrown?